torch/_inductor/codegen/wrapper.py

```diff
@@ -415,3 +494,112 @@ def call_kernel(self, name: str, kernel: Kernel):
 
     def writeline(self, line):
         self.lines.append(line)
+
+
+class CppWrapperCodeGen(WrapperCodeGen):
+    """
+    The outer wrapper that calls the kernels.
+    """
+
+    call_func_id = count()
+
+    def __init__(self):
+        self._call_func_id = next(CppWrapperCodeGen.call_func_id)
+        super().__init__()
+
+    def write_prefix(self):
+        self.prefix.splice(
+            """
+            async_compile.wait(globals())
+            del async_compile
+            from torch.utils.cpp_extension import load_inline
+            wrapper = (
+            '''
+            #include <dlfcn.h>
+            #include <assert.h>
+            """
+        )
+        with self.prefix.indent():
+            inp_len = len(V.graph.graph_inputs.keys())
+            output_refs = [x.codegen_reference() for x in V.graph.graph_outputs]
+            if output_refs:
+                if len(output_refs) == 1:
+                    output_types = "at::Tensor"
+                else:
+                    output_return_type = "at::Tensor"
+                    output_return_types = [output_return_type] * len(output_refs)
+                    output_return_types = ", ".join(output_return_types)
+                    output_types = f"std::tuple<{output_return_types}>"
+            else:
+                output_types = "void"
+
+            if inp_len != 0:
+                inputs_args = ["at::Tensor"] * len(V.graph.graph_inputs.keys())
+                inputs_args = ", ".join(inputs_args)
+                inputs_args = f"std::tuple<{inputs_args}>"
+
+                self.prefix.writeline(
+                    f"{output_types} call_{self._call_func_id}({inputs_args} args) {{"
+                )
+                inputs_keys_str = ", ".join(V.graph.graph_inputs.keys())
+                self.prefix.writeline(f"at::Tensor {inputs_keys_str};")
+                self.prefix.writeline(f"std::tie({inputs_keys_str}) = args;")
+            else:
+                self.prefix.writeline(
+                    f"{output_types} call_{self._call_func_id}(std::tuple<> args) {{"
+                )
+            for name in V.graph.randomness_seeds:
+                self.prefix.writeline(f"at::Tensor {name};")
+                self.prefix.writeline(
+                    f"{name} = at::randint(std::pow(2, 31), {{}}, at::ScalarType::Long);"
+                )
+            V.graph.sizevars.codegen(self.prefix, V.graph.graph_inputs)
+
+    def write_allocate_line(self, buffer):
+        self.writeline(CppAllocateLine(buffer))
+
+    def write_del_line(self, name):
+        return
```

Comment:
We need to free memory here

Reply:
Added `tensor.reset()` here for cpp wrapper.